frmts/miramon/miramon_rasterband.cpp

```diff
@@ -0,0 +1,1286 @@
+            {
+                // Before the minimum, we apply the color of the first
+                // element (as a placeholder).
+                AssignRGBColor(nIPaletteColor, 0);
+            }
+            else if (nIPaletteColor <= static_cast<int>(poBand->GetVisuMax()))
+            {
+                // Between the minimum and maximum, we apply the value
+                // read from the table.
+                if (static_cast<int>(eMMBytesPerPixel) < 2)
+                {
+                    // The value is applied directly.
+                    AssignRGBColor(nIPaletteColor, nFirstValidPaletteIndex);
+                    nFirstValidPaletteIndex++;
+                }
+                else
+                {
+                    // The value is applied according to the scaling.
+                    nIndexColor = static_cast<unsigned short>(
+                        dfSlope * nIPaletteColor + dfIntercept);
+                    if (nIndexColor > Palette->GetNumberOfColors())
+                        nIndexColor = static_cast<unsigned short>(
+                            Palette->GetNumberOfColors());
+                    AssignRGBColor(nIPaletteColor, nIndexColor);
+                }
+            }
+            else
+            {
+                // After the maximum, we apply the value of the last
+                // element (as a placeholder).
+                AssignRGBColor(nIPaletteColor,
+                               Palette->GetNumberOfColors() - 1);
+            }
+        }
+    }
+
+    return CE_None;
+}
+
+CPLErr MMRRasterBand::GetRATName(char *papszToken, CPLString &osRELName,
+                                 CPLString &osDBFName,
+                                 CPLString &osAssociateREL)
+{
+    CPLString os_Join = "JoinTaula";
+    os_Join.append("_");
+    os_Join.append(papszToken);
+
+    CPLString osTableNameSection_value;
+
+    if (!pfRel->GetMetadataValue(SECTION_ATTRIBUTE_DATA, osBandSection, os_Join,
+                                 osTableNameSection_value) ||
+        osTableNameSection_value.empty())
+        return CE_Failure;  // No attribute available
+
+    CPLString osTableNameSection = "TAULA_";
+    osTableNameSection.append(osTableNameSection_value);
+
+    CPLString osShortRELName;
+
+    if (!pfRel->GetMetadataValue(osTableNameSection, "NomFitxer",
+                                 osShortRELName) ||
+        osShortRELName.empty())
+    {
+        osRELName = "";
+        osAssociateREL = "";
+        return CE_Failure;
+    }
+
+    CPLString osExtension = CPLGetExtensionSafe(osShortRELName);
+    if (osExtension.tolower() == "rel")
+    {
+        // Get path relative to REL file
+        osRELName =
+            CPLFormFilenameSafe(CPLGetPathSafe(pfRel->GetRELNameChar()).c_str(),
+                                osShortRELName, "");
+
+        // Getting information from the associated REL
+        MMRRel *fLocalRel = new MMRRel(osRELName, false);
```

Comment:
no need for dynamic allocation here as far as I can see:
```suggestion
        MMRRel localRel(osRELName, false);
```